sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -0,0 +1,303 @@
+package org.apache.beam.sdk.io.gcp.spanner;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.client.util.BackOff;
+import com.google.api.client.util.BackOffUtils;
+import com.google.api.client.util.Sleeper;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.display.DisplayData;
+import org.apache.beam.sdk.transforms.display.DisplayData.Builder;
+import org.apache.beam.sdk.util.FluentBackoff;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * {@link SpannerIO} provides an API for reading from and writing to
+ * <a href="https://cloud.google.com/spanner/">Google Cloud Spanner</a>.
+ *
+ */
+@Experimental(Experimental.Kind.SOURCE_SINK)
+public class SpannerIO {
+
+  private SpannerIO() { }
+
+  @VisibleForTesting
+  public static final int SPANNER_MUTATIONS_PER_COMMIT_LIMIT = 20000;
+
+/*
+ * <p>To write a {@link PCollection} to Spanner, use {@link SpannerIO#writeTo} to
+ * specify the database location for output.
+ *
+ * <p>For example:
+ *
+ * <pre> {@code
+ * // Write data to a table.
+ * PipelineOptions options = PipelineOptionsFactory.fromArgs(args).create();
+ * options.getInput = <CSV text file location>
+ *
+ * Pipeline p = Pipeline.create(options);
+ *   p.apply(TextIO.Read.from(options.getInput()))
+ *       .apply(ParDo.of(new ParseLineFn(tableInfo)))
```

Comment:
Could you extract a variable here to make it explicit that it is a  `PCollection<Mutation>`